daskms/tests/test_github.py

```diff
@@ -0,0 +1,29 @@
+import os
+
+import dask
+import dask.array as da
+from daskms import xds_from_ms
+
+import pytest
+
+
+def test_github_98():
+    ms = "/home/sperkins/data/AF0236_spw01.ms/"
+
+    if not os.path.exists(ms):
+        pytest.skip("AF0236_spw01.ms on which this "
+                    "test depends is not present")
+
+    datasets = xds_from_ms(ms, columns=['DATA', 'ANTENNA1', 'ANTENNA2'],
+                           group_cols=['DATA_DESC_ID'],
+                           taql_where='ANTENNA1 == 5 || ANTENNA2 == 5')
+
+    assert len(datasets) == 2
+    assert datasets[0].DATA_DESC_ID == 0
+    assert datasets[1].DATA_DESC_ID == 1
+
+    for ds in datasets:
+        expr = da.logical_or(ds.ANTENNA1.data == 5, ds.ANTENNA2.data == 5)
+        expr, equal = dask.compute(expr, da.all(expr))
+        assert equal.item() is True
+        assert len(expr) > 0
```

Comment:
@IanHeywood Thanks for providing me with the MS for which you were encountering this problem. This test case failed for me prior to this commit by simply returning an empty iterator (and I believe reproducing @o-smirnov 's #100).

With the change in this commit, it does return two datasets (different DDID's but probably same SCAN/FIELD as the rows are the same) whose ANTENNA1 and ANTENNA2 values are different.

Reply:
Does this change fix the problem for you on your side? I couldn't reproduce your crash in this branch.